udata_hydra/utils/file.py

```diff
@@ -52,15 +52,18 @@ async def download_resource(
     async with aiohttp.ClientSession(
         headers={"user-agent": config.USER_AGENT}, raise_for_status=True
     ) as session:
-        async with session.get(url, allow_redirects=True) as response:
-            async for chunk in response.content.iter_chunked(chunk_size):
-                if max_size_allowed is None or i * chunk_size < max_size_allowed:
-                    tmp_file.write(chunk)
-                else:
-                    tmp_file.close()
-                    log.warning(f"File {url} is too big, skipping")
-                    raise IOError("File too large to download")
-                i += 1
+        try:
+            async with session.get(url, allow_redirects=True) as response:
+                async for chunk in response.content.iter_chunked(chunk_size):
+                    if max_size_allowed is None or i * chunk_size < max_size_allowed:
+                        tmp_file.write(chunk)
+                    else:
+                        tmp_file.close()
+                        log.warning(f"File {url} is too big, skipping")
+                        raise IOError("File too large to download")
+                    i += 1
+        except aiohttp.ClientResponseError as e:
+            raise IOError(f"Error downloading CSV: {e}")
```

Comment:
I don't think this makes the error more explicit than previously?
I think the issue is more about marking these as "expected" in sentry

Reply:
I think the error would be more explicit that way, if explained if the error title. For now in Sentry it's not easy to identify what is the issue is when listing the errors, so we can't see immediately it's not a major issue.

We could also create a specific exception which would include attributes to identify the ressource, as we did for the custom `ParseException`. What do you think?

Reply:
...Updated!